koschei/backend.py

```diff
@@ -209,6 +209,28 @@ def register_real_builds(self, task_info_mapping):
             if self.is_build_newer(pkg.last_build, info):
                 self.register_real_build(pkg, info)
 
+    def _set_blocked_status(self, packages, blocked):
+        if packages:
+            self.db.query(Package).filter(Package.id.in_(packages))\
+                   .update({'blocked': blocked}, synchronize_session=False)
+            self.db.expire_all()
+            self.db.flush()
+
+    def refresh_blocked(self):
+        """
+        Checks Koji for blocked and unblocked packages
+        """
+        packages = self.db.query(Package).all()
+        for p in packages:
+            self.db.expunge(p)
+        source_tag = util.koji_config['source_tag']
+        whitelisted = {p['package_name'] for p in
+                       self.koji_session.listPackages(tagID=source_tag, inherited=True)
+                       if not p['blocked']}
+        to_update = [p for p in packages if p.blocked == p.name in whitelisted]
+        self._set_blocked_status({p.id for p in to_update if p.blocked}, False)
+        self._set_blocked_status({p.id for p in to_update if not p.blocked}, True)
```

Comment:
If I understand this correctly, then you want to negate `blocked` of packages in `to_update`. So I believe you can do that in one query:
````python
self.db.query(Package)\
           .filter(Package.id.in_(p.id for p in to_update))\
           .update({'blocked': ~Package.blocked}, synchronize_session=False)
````
That way you don't need the auxiliary function and also don't need to expunge.